File: java/src/test/java/com/alxarafe/domain/rules/valueobject/RuleTypeTest.java
package com.alxarafe.domain.rules.valueobject;

import com.alxarafe.app.domain.rules.valueobject.RuleType;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class RuleTypeTest {

    @Test
    void values() {
        assertEquals(RuleType.REQUIRES, RuleType.valueOf("REQUIRES"));
        assertEquals(RuleType.FORBIDS, RuleType.valueOf("FORBIDS"));
        assertEquals(2, RuleType.values().length);
    }
}

File: php/tests/Unit/Domain/Rules/ValueObject/RuleTypeTest.php
<?php

declare(strict_types=1);

namespace Tests\Unit\Domain\Rules\ValueObject;

use Alxarafe\App\Domain\Rules\ValueObject\RuleType;
use PHPUnit\Framework\TestCase;

final class RuleTypeTest extends TestCase
{
    public function testValues(): void
    {
        self::assertSame('REQUIRES', RuleType::REQUIRES->value);
        self::assertSame('FORBIDS', RuleType::FORBIDS->value);
    }
}
